tests/unit/test_record_check.py

```diff
@@ -0,0 +1,153 @@
+# -*- coding: utf-8 -*-
```

Comment:
Can you include a test around what happens if `ProxyResolver` raises an exception?